crates/bin/pd/src/main.rs

```diff
@@ -458,6 +458,12 @@ async fn main() -> anyhow::Result<()> {
                 )));
             }
 
+            // TODO: will need a bit of finesse to make this work with auto-https,
+            // since auto-https support above works with the tonic `serve_with_incoming`
+            // method, but now we're dropping down to axum directly. just needs some realignment
+            // probably.
```

Comment:
In the ~year since this code was written, there may be better options. https://docs.rs/tokio-rustls-acme/latest/tokio_rustls_acme/ seems promising